### ISSUE TITLE
[false-negative]ToStringReturnsNull can not detect the case return condition ? foo : null

### ISSUE THREAD

**Issue by wuchiuwong** [google/error-prone#1659]
### Description of the problem / feature request:

ToStringReturnsNull can not detect the case <code>return condition ? foo : null</code>
This rule is implemented in error_prone_core-2.3.4-sources.jar!\com\google\errorprone\bugpatterns\ToStringReturnsNull.java, 
and use <code>node.getExpression().getKind() == NULL_LITERAL;</code> to check whether null is returned,
Therefore, it will not detect those return null statements that make logical judgments when returning.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

```
@Override
public String toString() {
	return this.mode == 'd' ? this.writer.toString() : null;
}
```

### What version of Error Prone are you using?

2.3.4

###  Have you found anything relevant by searching the web?

no

**Comment by sumitbhagwani** [comment]
There are numerous cases where toString() implementation can return null. 
ToStringReturnsNull check focuses on identifying specific cases where toString() will return null rather than exhaustively looking for all such cases.
